pkg/diagnostics/v1/types.go

```diff
@@ -0,0 +1,301 @@
+		ClusterService string `json:"cluster_service"`
+		Host           string `json:"host"`
+		Location       string `json:"location"`
+		Name           string `json:"name"`
+		Prefix         string `json:"prefix"`
+		Rewrite        string `json:"rewrite"`
+	} `json:"mappings"`
+	MetadataLabels struct {
+		AmbassadorCrd           string `json:"ambassador_crd"`
+		AppKubernetesIoInstance string `json:"app.kubernetes.io/instance"`
+	} `json:"metadata_labels"`
+	Name            string        `json:"name"`
+	Namespace       string        `json:"namespace"`
+	Precedence      int           `json:"precedence"`
+	Prefix          string        `json:"prefix"`
+	QueryParameters []interface{} `json:"query_parameters"`
+	RegexRewrite    struct{}      `json:"regex_rewrite"`
+	Rewrite         string        `json:"rewrite"`
+	TimeoutMs       int           `json:"timeout_ms"`
+	Serialization   string        `json:"serialization"`
+}
+
+type Loginfo struct {
+	All string `json:"all"`
+}
+
+type Notice struct {
+	Level   string `json:"level"`
+	Message string `json:"message"`
+}
+
+type System struct {
+	AmbassadorID        string        `json:"ambassador_id"`
+	AmbassadorNamespace string        `json:"ambassador_namespace"`
+	BootTime            string        `json:"boot_time"`
+	ClusterID           string        `json:"cluster_id"`
+	DebugMode           bool          `json:"debug_mode"`
+	EndpointsEnabled    bool          `json:"endpoints_enabled"`
+	EnvFailures         []interface{} `json:"env_failures"`
+	EnvGood             bool          `json:"env_good"`
+	EnvStatus           struct {
+		ErrorCheck struct {
+			Specifics [][]interface{} `json:"specifics"`
+			Status    bool            `json:"status"`
+		} `json:"Error check"`
+		Mappings struct {
+			Specifics [][]interface{} `json:"specifics"`
+			Status    bool            `json:"status"`
+		} `json:"Mappings"`
+		TLS struct {
+			Specifics [][]interface{} `json:"specifics"`
+			Status    bool            `json:"status"`
+		} `json:"TLS"`
+	} `json:"env_status"`
+	Hostname        string `json:"hostname"`
+	HrUptime        string `json:"hr_uptime"`
+	KnativeEnabled  bool   `json:"knative_enabled"`
+	LatestSnapshot  string `json:"latest_snapshot"`
+	SingleNamespace bool   `json:"single_namespace"`
+	StatsdEnabled   bool   `json:"statsd_enabled"`
+	Version         string `json:"version"`
+}
+type TLSContext struct {
+	Active       bool     `json:"_active"`
+	Errored      bool     `json:"_errored"`
+	ReferencedBy []string `json:"_referenced_by"`
+	Rkey         string   `json:"_rkey"`
+	Hosts        []string `json:"hosts"`
+	IsFallback   bool     `json:"is_fallback"`
+	Kind         string   `json:"kind"`
+	Location     string   `json:"location"`
+	Name         string   `json:"name"`
+	Namespace    string   `json:"namespace"`
+	SecretInfo   struct {
+		CertChainFile  string `json:"cert_chain_file"`
+		PrivateKeyFile string `json:"private_key_file"`
+		Secret         string `json:"secret"`
+	} `json:"secret_info"`
+}
```

Comment:
Amazing work at writing this structure.
I see `hColor` , `service` , and `weight`, `notices` But I can't find `key`.
As in `"key": "http://staging-app.datawire.io/download/tel2/windows/amd64/nightly/telepresence.zip",`
I think it should be in the Groups struct. (Note, "key" is different from `rKey` and `cache_key`